aiida/manage/configuration/migrations/migrations.py

```diff
@@ -296,6 +296,53 @@ def downgrade(self, config: ConfigType) -> None:
                 CONFIG_LOGGER.warning(f'profile {profile_name!r} had no expected "storage._v6_backend" key')
 
 
+class AddTestProfileKey(SingleMigration):
+    """Add the ``test_profile`` key."""
+    down_revision = 7
+    down_compatible = 7
+    up_revision = 8
+    up_compatible = 8
+
+    def upgrade(self, config: ConfigType) -> None:
+        for profile_name, profile in config.get('profiles', {}).items():
+            profile['test_profile'] = profile_name.startswith('test_')
+
+    def downgrade(self, config: ConfigType) -> None:
+        profiles = config.get('profiles', {})
+        profile_names = list(profiles.keys())
+
+        # Iterate over the fixed list of the profile names, since we are mutating the profiles dictionary.
+        for profile_name in profile_names:
```

Comment:
Nice! I see you are still pushing changes, so ping me when you want the final review.

Reply:
I am done. Everything is addressed and commits are rebased